src/core/query-builder.constants.ts

```diff
@@ -0,0 +1,201 @@
+    expressionTemplate: '{0}.Contains("{1}")',
+  },
+  LIST_DOES_NOT_EQUAL: {
+    label: 'Does not equal',
+    name: 'listnotequals',
+    expressionTemplate: '!({0}.Contains("{1}"))',
+  },
+  LIST_CONTAINS: {
+    label: 'Contains',
+    name: 'listcontains',
+    expressionTemplate: '{0}.Any(it.Contains("{1}"))',
+  },
+  LIST_DOES_NOT_CONTAIN: {
+    label: 'Does not contain',
+    name: 'listnotcontains',
+    expressionTemplate: '{0}.Any(!it.Contains("{1}"))',
+  },
+  // Properties expressions
+  PROPERTY_EQUALS: {
+    label: 'Equals',
+    name: 'propertyequals',
+    expressionTemplate: 'properties["{0}"] = "{1}"',
+  },
+  PROPERTY_DOES_NOT_EQUAL: {
+    label: 'Does not equal',
+    name: 'propertynotequals',
+    expressionTemplate: 'properties["{0}"] != "{1}"',
+  },
+  PROPERTY_STARTS_WITH: {
+    label: 'Starts with',
+    name: 'propertystartswith',
+    expressionTemplate: 'properties["{0}"].StartsWith("{1}")',
+  },
+  PROPERTY_ENDS_WITH: {
+    label: 'Ends with',
+    name: 'propertyendswith',
+    expressionTemplate: 'properties["{0}"].EndsWith("{1}")',
+  },
+  PROPERTY_CONTAINS: {
+    label: 'Contains',
+    name: 'propertycontains',
+    expressionTemplate: 'properties["{0}"].Contains("{1}")',
+  },
+  PROPERTY_DOES_NOT_CONTAIN: {
+    label: 'Does not contains',
+    name: 'propertynotcontains',
+    expressionTemplate: '!(properties["{0}"].Contains("{1}"))',
+  },
+  PROPERTY_IS_BLANK: {
+    label: 'Is blank',
+    name: 'propertyisblank',
+    expressionTemplate: 'string.IsNullOrEmpty(properties["{0}"])',
+    hideValue: true,
+  },
+  PROPERTY_IS_NOT_BLANK: {
+    label: 'Is not blank',
+    name: 'propertyisnotblank',
+    expressionTemplate: '!string.IsNullOrEmpty(properties["{0}"])',
+    hideValue: true,
+  },
+}
+
+export const customOperations: QueryBuilderCustomOperation[] = [
+  QueryBuilderOperations.EQUALS,
+  QueryBuilderOperations.DOES_NOT_EQUAL,
+  QueryBuilderOperations.STARTS_WITH,
+  QueryBuilderOperations.ENDS_WITH,
+  QueryBuilderOperations.CONTAINS,
+  QueryBuilderOperations.DOES_NOT_CONTAIN,
+  QueryBuilderOperations.IS_BLANK,
+  QueryBuilderOperations.IS_NOT_BLANK,
+  QueryBuilderOperations.GREATER_THAN,
+  QueryBuilderOperations.GREATER_THAN_OR_EQUAL_TO,
+  QueryBuilderOperations.LESS_THAN,
+  QueryBuilderOperations.LESS_THAN_OR_EQUAL_TO,
+  QueryBuilderOperations.LIST_EQUALS,
+  QueryBuilderOperations.LIST_DOES_NOT_EQUAL,
+  QueryBuilderOperations.LIST_CONTAINS,
+  QueryBuilderOperations.LIST_DOES_NOT_CONTAIN,
+  QueryBuilderOperations.PROPERTY_EQUALS,
+  QueryBuilderOperations.PROPERTY_DOES_NOT_EQUAL,
+  QueryBuilderOperations.PROPERTY_STARTS_WITH,
+  QueryBuilderOperations.PROPERTY_ENDS_WITH,
+  QueryBuilderOperations.PROPERTY_CONTAINS,
+  QueryBuilderOperations.PROPERTY_DOES_NOT_CONTAIN,
+  QueryBuilderOperations.PROPERTY_IS_BLANK,
+  QueryBuilderOperations.PROPERTY_IS_NOT_BLANK,
+];
+
+export enum ComputedDataFields {
+  Location = 'Location.MinionId || Location.PhysicalLocation'
+}
```

Comment:
This should be a const object or an array, not an enum. I also don't think it should be in this file. `setQueryComputedFields()` should accept this object/array as an parameter and then the asset specific fields should be defined in the asset data sources.

Reply:
Created separate object that will hold required transformation, please check the new approach.